File: CodeRun_Boost/7165_coderun_welcome/solution.js
function check(n, m, k) {
    // All numbers in JS are 64-bit floats, which can safely handle the integer math here.
    return (k * k <= n + m) && (Math.floor((k * k) / 2) <= Math.min(n, m));
}

function solution(n, m) {
    let left = 0;
    let right = 150000;
    let ans = 0;

    while (left <= right) {
        let mid = Math.floor(left + (right - left) / 2);

        if (check(n, m, mid)) {
            ans = mid;
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return ans;
}

// Export the solution function for Node.js environments
module.exports = solution;

File: CodeRun_Boost/7165_coderun_welcome/solution.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool check(long long n, long long m, long long k) {
    return (k * k <= n + m) && ((k * k) / 2 <= (n < m ? n : m));
}

long long solution(long long n, long long m) {
    long long left = 0;
    long long right = 150000;

    long long ans = 0;

    while (left <= right) {
        long long mid = left + (right - left) / 2;

        if (check(n, m, mid)) {
            ans = mid;
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return ans;
}